### PR TITLE
docs(vault): add env vault configuration options in kong.conf

### PR DESCRIPTION
**Pull request by bungle**
### Summary

Adds docs about `env` vault configuration options.

[KAG-1203](https://konghq.atlassian.net/browse/KAG-1203)

[KAG-1203]: https://konghq.atlassian.net/browse/KAG-1203?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: kong.conf.default [hunk 1]
```diff
@@ -1587,6 +1587,33 @@
...
                                  # single query.
 
+#------------------------------------------------------------------------------
+# VAULTS
+#------------------------------------------------------------------------------
+
+# A secret is any sensitive piece of information required for API gateway
+# operations. Secrets may be part of the core Kong Gateway configuration,
+# they may be used in plugins, or they might be part of configuration
+# associated with APIs serviced by the gateway.
```
Comment (jschmid1) [inline kong.conf.default:1597]: How about: 
```suggestion
# operations. Secrets may be part of the core Kong Gateway configuration, used in plugins, or part of the configuration associated with APIs serviced by the gateway.
```

File: kong.conf.default [hunk 2]
```diff
@@ -1587,6 +1587,33 @@
...
+#
+# Some of the most common types of secrets used by Kong Gateway include:
+#
+# - Data store usernames and passwords, used with PostgreSQL and Redis
+# - Private X.509 certificates
+# - API keys
+#
+# Sensitive plugin configuration fields, generally used for authentication,
+# hashing, signing, or encryption. Kong Gateway lets you store certain values
```
Comment (jschmid1) [inline kong.conf.default:1606]:
```suggestion
# Sensitive plugin configuration fields are generally used for authentication, hashing, signing, or encryption. Kong Gateway lets you store certain values
```

**Review by jschmid1 — APPROVED** [review]
Sounds sane. Left my 2ct

File: kong.conf.default [hunk 3]
```diff
@@ -1587,6 +1587,33 @@
...
+# Sensitive plugin configuration fields are generally used for authentication,
+# hashing, signing, or encryption. Kong Gateway lets you store certain values
+# in a vault. Here are the vault specific configuration options.
+
+#vault_env_prefix =              # Defines the environment variable vault's
+                                 # default prefix. For example if you have
+                                 # all your secrets stored in environment
+                                 # variables prefixed with `SECRETS_`, it
+                                 # can be configured here so that it isn't
+                                 # necessary to repeat them on Vault
```
Comment (Guaris) [inline kong.conf.default:1614]:
```suggestion
                                 # necessary to repeat them in Vault
```

**Pull request merged**